### ISSUE TITLE
TypeError when running example plan and version 0.8.0

### ISSUE THREAD

**Issue by lucasasmith** [Titan-Systems/titan#76]
When running the [example](https://titan-core.gitbook.io/titan-core/getting-started#using-the-python-package) with version `0.8.0` I get the below error.

![image](https://github.com/user-attachments/assets/7f87c905-1db2-4157-834a-0a04a777bf14)

When I `pip uninstall titan-core` and `pip install titan-core==0.7.4`, the plan works.
```
» titan core
» Plan: 3 to add, 0 to change, 0 to destroy.
```

BTW, Titan looks very nice, looking forward to testing more.

**Comment by toadies** [comment]
I get the same error.  It appears to be related to `GrantedPrivilege.from_grant` and `PRIVS_FOR_RESOURCE_TYPE` returning None, resulting in a NoneType error not ValueType.

```
granted_priv = GrantedPrivilege.from_grant(
                    privilege="READ",
                    granted_on="IMAGE REPOSITORY",
                    name="SNOWHOUSE_IMPORT.IMAGES.SNOWFLAKE_IMAGES",
                )

Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "<stdin>", line 9, in from_grant
TypeError: 'NoneType' object is not callable
```

To fix this, I am manually raising ValueError
```
@dataclass
class GrantedPrivilege:
    privilege: Priv
    on: str  # probably should be FQN

    @classmethod
    def from_grant(cls, privilege: str, granted_on: str, name: str):
        resource_type = ResourceType(granted_on)
        priv_type = PRIVS_FOR_RESOURCE_TYPE[resource_type]
        
        if priv_type is None:
            raise ValueError(f"{resource_type} has no valid privilege resource")
        
        priv = priv_type(privilege)
        return cls(privilege=priv, on=name)
```

Print out:
Privileges for type: IMAGE_REPOSITORY not currently supported: IMAGE REPOSITORY has no valid privilege resource

Cheers

**Comment by teej** [comment]
Fixed in [v0.8.1](https://github.com/Titan-Systems/titan/releases/tag/v0.8.1). If you have any other questions or issues let me know!

**Comment by toadies** [comment]
Looks good! Thanks
